src/main.rs

```diff
@@ -15,6 +33,124 @@ struct Args {
     settings_file: String,
 }
 
+fn test_and_run_loop() {
+    let mut settings: HashMap<i32, Settings> = HashMap::default();
+    let mut effects: HashMap<i32, Effect> = HashMap::default();
+    let mut effect_settings: HashMap<i32, i32> = HashMap::default();
+    let mut connections: HashMap<i32, Connection> = HashMap::default();
+    let mut ledstrips = vec![];
+
+    let moody_settings = MoodySettings {
+        color: Color { r: 255, g: 0, b: 0 },
+    };
+    let raindrop_settings = RaindropSettings {
+        rain_speed: 1,
+        drop_rate: 0.10,
+    };
+    settings.insert(0, Settings::Moody(moody_settings));
+    settings.insert(1, Settings::Raindrop(raindrop_settings));
+
+    let moody = Moody { id: 10 };
+    effects.insert(10, Effect::Moody(moody));
+    effect_settings.insert(10, 0);
+
+    let raindrop = Raindrops {
+        id: 20,
+        state: RaindropState { riples: vec![] },
+    };
+    effects.insert(20, Effect::Raindrop(raindrop));
+    effect_settings.insert(20, 1);
+
+    let ip = std::net::SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(192, 168, 0, 197), 1234));
+    let connection = TcpConnection::new(ip);
+    let connection_id = 1;
+    connections.insert(connection_id, Connection::Tcp(connection));
+    connections.insert(2, Connection::Usb(UsbConnection {}));
+
+    let mut ls1 = LedStrip::new();
+    ls1.set_led_count(300);
+    ls1.add_effect(20, 300);
+    ls1.connection_id = Some(connection_id);
+    ledstrips.push(ls1);
+
+    loop {
+        std::thread::sleep(std::time::Duration::from_millis(16));
+        tick(
+            &mut ledstrips,
+            &mut effects,
+            &mut settings,
+            &effect_settings,
+            &mut connections,
+        );
+    }
+}
+
+fn send_to_connection(ledstrip: &mut LedStrip, connections: &mut HashMap<i32, Connection>) {
+    if ledstrip.connection_id.is_none() {
+        return;
```

Comment:
tu devrais plus utiliser les Results, comme ca ici tu pourrais juste `let connection_id = ledstrip.connection_id?;`

Reply:
et aussi comme ca on peut avoir du meilleur error handling et du meilleur logging

Reply:
et aussi on devrait utiliser rust nightly pcq comme ca on pourrait utiliser `let else` (https://www.youtube.com/watch?v=U-5_bumwH9w)
au lieu de faire comme tu fais souvent `if is_none ... unwrap`

Reply:
I agree pour le error handdling. Ici par contre je crois qu'on voudrait juste early-return pcq cest le expected behavior imo. Ta strip a pas de connection so juste early return no need to send an error.

Reply:
tu peux faire ca avec les options `let connection_id = ledstrip.connection_id?;`

Reply:
Jsuis pas contre le option, mais je crois que si ya pas de connection on devrait return kkch pour le dire a la personne qui call